prow/spyglass/gcsartifact_fetcher_test.go

```diff
@@ -37,7 +38,7 @@ func TestNewGCSJobSource(t *testing.T) {
 	}{
 		{
 			name:        "Test standard GCS link",
-			src:         "test-bucket/logs/example-ci-run/403",
+			src:         "gs://test-bucket/logs/example-ci-run/403",
```

Comment:
Please leave the existing tests, they have to continue passing due to compatibility concerns and add new ones

Reply:
`newGCSJobSource` did only handle the new format with `<storage-provider>://<path>` and was only used like this. But I extended `newGCSJobSource` and the tests to handle old and new format